include/deal.II/grid/manifold_lib.h

```diff
@@ -593,12 +593,16 @@ class FunctionManifold : public ChartManifold<dim, spacedim, chartdim>
    *
    * The tolerance argument is used in debug mode to actually check that the
    * two functions are one the inverse of the other.
+   *
+   * By default, only pointers to the passed functions are stored.
+   * Use @p owns_functions to change this behaviour.
    */
   FunctionManifold(
     const Function<chartdim> & push_forward_function,
     const Function<spacedim> & pull_back_function,
-    const Tensor<1, chartdim> &periodicity = Tensor<1, chartdim>(),
-    const double               tolerance   = 1e-10);
+    const Tensor<1, chartdim> &periodicity    = Tensor<1, chartdim>(),
+    const double               tolerance      = 1e-10,
+    bool                       owns_functions = false);
```

Comment:
Oh OK so that's how you fixed the memory leak from python. This works with me but I don't really know anything about the `Manifold` interface. @dealii/dealii Can someone who knows more about the `Manifold` interface give an opinion on this.